dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_configurations_panel.tsx

```diff
@@ -118,12 +129,22 @@ export const DataConfigPanelItem = ({
     setConfigList(updatedList);
   };
 
+  /**
+   * Removes element from config list
+   * @param index position of element to be removed
+   * @param name section name for which cross icon is clicked
+   */
   const handleServiceRemove = (index: number, name: string) => {
-    const list = { ...configList };
-    const arr = [...list[name]];
-    arr.splice(index, 1);
-    const updatedList = { ...list, [name]: arr };
-    setConfigList(updatedList);
+    let list = { ...configList };
+    const isTimeStamp = name === SPAN;
+    if (isTimeStamp) {
+      delete list[SPAN];
+    } else {
+      const arr = [...list[name]];
+      arr.splice(index, 1);
+      list = { ...list, [name]: arr };
```

Comment:
Take this as a future task to create an utility function to insert item at a specific position. I see this type of code so many times, those can be updated with this utility function. Make sure to de-structure the elements inside the function and not while calling it. You call the function with an array and the function is going to return you a new array using the supplied list.

Reply:
will take it in next PR and change all across the code where same logic has been implemented